CondTools/RunInfo/src/LHCInfoPerFillPopConSourceHandler.cc

```diff
@@ -0,0 +1,770 @@
+        std::make_pair("PB82", LHCInfoPerFill::PB82),
+        std::make_pair("AR18", LHCInfoPerFill::AR18),
+        std::make_pair("D", LHCInfoPerFill::D),
+        std::make_pair("XE54", LHCInfoPerFill::XE54)};
+
+    LHCInfoPerFill::FillType fillTypeFromString(const std::string& s_fill_type) {
+      for (auto const& i : s_fillTypeMap)
+        if (s_fill_type == i.first)
+          return i.second;
+      return LHCInfoPerFill::UNKNOWN;
+    }
+
+    LHCInfoPerFill::ParticleType particleTypeFromString(const std::string& s_particle_type) {
+      for (auto const& i : s_particleTypeMap)
+        if (s_particle_type == i.first)
+          return i.second;
+      return LHCInfoPerFill::NONE;
+    }
+  }  // namespace LHCInfoPerFillPopConImpl
+
+  namespace impl {
+
+    template <>
+    LHCInfoPerFill::FillType from_string(const std::string& attributeValue) {
+      return from_string_impl<LHCInfoPerFill::FillType, &LHCInfoPerFillPopConImpl::fillTypeFromString>(
+          attributeValue, LHCInfoPerFill::UNKNOWN);
+    }
+
+    template <>
+    LHCInfoPerFill::ParticleType from_string(const std::string& attributeValue) {
+      return from_string_impl<LHCInfoPerFill::ParticleType, &LHCInfoPerFillPopConImpl::particleTypeFromString>(
+          attributeValue, LHCInfoPerFill::NONE);
+    }
+
+  }  // namespace impl
+}  // namespace cond
+
+LHCInfoPerFillPopConSourceHandler::LHCInfoPerFillPopConSourceHandler(edm::ParameterSet const& pset)
+    : m_debug(pset.getUntrackedParameter<bool>("debug", false)),
+      m_startTime(),
+      m_endTime(),
+      m_samplingInterval((unsigned int)pset.getUntrackedParameter<unsigned int>("samplingInterval", 300)),
+      m_endFillMode(pset.getUntrackedParameter<bool>("endFill", true)),
+      m_name(pset.getUntrackedParameter<std::string>("name", "LHCInfoPerFillPopConSourceHandler")),
+      m_connectionString(pset.getUntrackedParameter<std::string>("connectionString", "")),
+      m_ecalConnectionString(pset.getUntrackedParameter<std::string>("ecalConnectionString", "")),
+      m_dipSchema(pset.getUntrackedParameter<std::string>("DIPSchema", "")),
+      m_authpath(pset.getUntrackedParameter<std::string>("authenticationPath", "")),
+      m_omsBaseUrl(pset.getUntrackedParameter<std::string>("omsBaseUrl", "")),
+      m_fillPayload(),
+      m_prevPayload(),
+      m_tmpBuffer() {
+  if (pset.exists("startTime")) {
+    m_startTime = boost::posix_time::time_from_string(pset.getUntrackedParameter<std::string>("startTime"));
+  }
+  boost::posix_time::ptime now = boost::posix_time::second_clock::local_time();
+  m_endTime = now;
+  if (pset.exists("endTime")) {
+    m_endTime = boost::posix_time::time_from_string(pset.getUntrackedParameter<std::string>("endTime"));
+    if (m_endTime > now)
+      m_endTime = now;
+  }
+}
+//L1: try with different m_dipSchema
+//L2: try with different m_name
+LHCInfoPerFillPopConSourceHandler::~LHCInfoPerFillPopConSourceHandler() {}
+
+namespace LHCInfoPerFillImpl {
```

Comment:
```suggestion
namespace theLHCInfoPerFillImpl {
```
rule 2.7 https://cms-sw.github.io/cms_coding_rules.html